openfe/setup/methods/rbfe_methods.py

```diff
@@ -0,0 +1,189 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""Contains RBFE methods
+
+
+"""
+from __future__ import annotations
+
+from openfe.setup import LigandAtomMapping, LigandMolecule
+from openfe.setup.methods import FEMethod
+from typing import Dict, Union
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+
+class ForcefieldSettings(BaseModel):
+    class Config:
+        arbitrary_types_allowed = True
+
+    # mapping of component name to forcefield path
+    forcefield: Dict[str, str]
+    nonbonded_electostatics = 'PME'
+    nonbonded_cutoff = 0.9 * unit.nanometer
+    constraints = 'HBonds'
+    rigid_water = True
+
+
+class StatePoint(BaseModel):
+    class Config:
+        arbitrary_types_allowed = True
+
+    temperature = 298.15 * unit.kelvin
+    pressure = 1 * unit.bar
+
+    # TODO: Validate that the units are correct (i.e. a pressure unit for pressure)
+    @validator('pressure', 'temperature')
+    def must_be_positive(cls, v):
+        if v <= 0:
+            raise ValueError("Must be positive")
+        return v
+
+
+# maybe this shouldn't be a pydantic class and instead our lambda protocol object
+class LambdaProtocol(BaseModel):
+    functions = 'default'
+    windows = 11
+    sample_endstates = False
+
+
+class OtherBarostat(BaseModel):
+    class Config:
+        extra = 'forbid'
+
+    freq = 10
+    thing = 20
+
+
+class OpenMM_MonteCarloBarostat(BaseModel):
+    class Config:
+        extra = 'forbid'
+
+    frequency = 50  # TODO: Use timestep unit?
```

Comment:
Having a time step unit will help make it clear that frequency isn't supposed to be a unit of time (like ps or fs)

Reply:
Not sure I understand what a "timestep unit" means, just calling it "steps"?

Reply:
It's to try and avoid confusing `ps` and `steps`.  The nice thing about units & pydantic validation is it will raise an error if you try and give a `500 * unit.timestep` to something expecting a `ps` unit variable

Reply:
Yeah sorry I meant more that I couldn't fully parse what Mike was proposing - the `unit.timestep` was my interpretation (which I agree with), I just wasn't sure if that was what was being suggested.

Reply:
Yes sorry for the confusion!  `unit.timestep` and `unit.seconds` will be distinct so we can use pydantic to help users from confialting them